llarp/iwp/session.cpp

```diff
@@ -822,20 +817,33 @@ namespace llarp
       {
         if (itr->second.Verify())
         {
-          auto msg = std::move(itr->second);
-          const llarp_buffer_t buf(msg.m_Data);
-          m_Parent->HandleMessage(this, buf);
-          if (m_ReplayFilter.emplace(itr->first, m_Parent->Now()).second)
-            m_SendMACKs.emplace(itr->first);
+          HandleRecvMsgCompleted(itr->second);
         }
         else
         {
           LogError("hash mismatch for message ", itr->first);
         }
-        m_RXMsgs.erase(itr);
       }
     }
 
+    void
+    Session::HandleRecvMsgCompleted(const InboundMessage& msg)
+    {
+      /// should we send a multiack?
+      constexpr bool UseMACK = false;
```

Comment:
if getting rid of multi-ack, should probably get rid of conditionals on it.  can always revert if we change our minds.

Reply:
done